Make LengthSize and LengthPoint consistent

NathanSolomon733 opened the pull request:
#### 42049e8173df70e36fdcffcf2369f8de82b5b7c1
<pre>
Make LengthSize and LengthPoint consistent
<a href="https://bugs.webkit.org/show_bug.cgi?id=277521">https://bugs.webkit.org/show_bug.cgi?id=277521</a>
<a href="https://rdar.apple.com/133228272">rdar://133228272</a>

Reviewed by Tim Nguyen.

This fix  gets rid of using public/private control in LengthPoint, which was already a struct. It makes all members of LengthPoint public and removes their getter and setter members.

* Source/WebCore/animation/CSSPropertyAnimation.cpp:
* Source/WebCore/css/BasicShapeConversion.cpp:
(WebCore::valueForBasicShape):
* Source/WebCore/css/BasicShapesShapeSegmentConversion.cpp:
(WebCore::lengthPointToCSSValue):
* Source/WebCore/css/ComputedStyleExtractor.cpp:
(WebCore::valueForPosition):
(WebCore::isAuto):
(WebCore::isNormal):
(WebCore::valueForPathOperation):
* Source/WebCore/platform/LengthFunctions.cpp:
(WebCore::pointForLengthPoint):
(WebCore::floatPointForLengthPoint):
* Source/WebCore/platform/LengthPoint.cpp:
(WebCore::operator&lt;&lt;):
* Source/WebCore/platform/LengthPoint.h:
(WebCore::LengthPoint::LengthPoint):
(WebCore::LengthPoint::isZero const):
(WebCore::blend):
(WebCore::LengthPoint::setX): Deleted.
(WebCore::LengthPoint::x const): Deleted.
(WebCore::LengthPoint::setY): Deleted.
(WebCore::LengthPoint::y const): Deleted.
* Source/WebCore/platform/animation/AcceleratedEffectValues.cpp:
(WebCore::nonCalculatedLengthPoint):
(WebCore::AcceleratedEffectValues::AcceleratedEffectValues):
* Source/WebCore/rendering/MotionPath.cpp:
(WebCore::MotionPath::motionPathDataForRenderer):
(WebCore::MotionPath::applyMotionPathTransform):
* Source/WebCore/rendering/RenderReplaced.cpp:
(WebCore::RenderReplaced::paint):
* Source/WebCore/rendering/style/ShadowData.cpp:
(WebCore::ShadowData::ShadowData):
* Source/WebCore/rendering/style/ShadowData.h:
(WebCore::ShadowData::ShadowData):
(WebCore::ShadowData::x const):
(WebCore::ShadowData::y const):
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

Canonical link: <a href="https://commits.webkit.org/284009@main">https://commits.webkit.org/284009@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4b49aeb33a117a5f6d8ca88bbda50b928e3d2269

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/68042 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/47434 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/20701 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/72101 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/19187 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/70159 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/55230 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/19003 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/54387 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/12796 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/71109 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/43436 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/58800 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/34849 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/40105 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/16216 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/17544 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/62079 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/16557 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/73801 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/12013 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/15836 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/61840 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/12051 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/58875 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/61856 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/15122 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/9757 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/3370 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/43234 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/44308 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/45503 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/44050 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

smfr commented on Source/WebCore/platform/LengthPoint.h:
```diff
@@ -32,31 +32,12 @@ namespace WebCore {
-
-private:
     // FIXME: it would be nice to pack the two Lengths together better somehow (to avoid padding between them).
     Length m_x;
```
Our coding conventions suggest that structure members don't have the `m_` prefix

NathanSolomon733 replied:
Understood. I'll fix that.

brentfulgham commented on Source/WebCore/platform/LengthPoint.h:
```diff
@@ -32,36 +32,17 @@ namespace WebCore {
-        : m_x(WTFMove(x))
-        , m_y(WTFMove(y))
-    {
-    }
```
The `Length` structure consists of a union (packed into an integer), an 8-bit type, and two bools. The copy operation is probably super fast, so these move operations were never useful.

nt1m commented on Source/WebCore/platform/LengthPoint.h:
```diff
@@ -32,36 +32,27 @@ namespace WebCore {
+        : x(x)
+        , y(y)
     {
     }
```
Do we actually need these constructors? Pretty sure they're default.

NathanSolomon733 replied:
We do not. The only slight benefit that this has is that if another class has a LengthPoint member variable in it, I believe you would need to use braces instead of parentheses in an initializer list such as: `m_lengthPoint{x, y}` instead of `m_lengthPoint(x, y)`. The initialization with the parentheses would try to call a constructor that specifically takes 2 parameters. I removed them previously and got a compilation error because of this.
I asked Matthieu about his preference and he said to just keep them.

nt1m commented on Source/WebCore/platform/LengthPoint.h:
```diff
@@ -32,36 +32,27 @@ namespace WebCore {
     {
     }
-    
+
```
Could we eliminate this constructor?

NathanSolomon733 replied:
We need to keep this default constructor since other instances of LengthPoint are already using it. The compiler will not generate this constructor automatically for us if we also provide the constructor LengthPoint(x, y) unless we keep this or we write: `LengthPoint() = default; `

nt1m replied:
`LengthPoint() = default;` is more elegant

NathanSolomon733 replied:
Done.

NathanSolomon733 committed:
Make LengthSize and LengthPoint consistent
https://bugs.webkit.org/show_bug.cgi?id=277521
rdar://133228272

Reviewed by Tim Nguyen.

This fix  gets rid of using public/private control in LengthPoint, which was already a struct. It makes all members of LengthPoint public and removes their getter and setter members.

* Source/WebCore/animation/CSSPropertyAnimation.cpp:
* Source/WebCore/css/BasicShapeConversion.cpp:
(WebCore::valueForBasicShape):
* Source/WebCore/css/BasicShapesShapeSegmentConversion.cpp:
(WebCore::lengthPointToCSSValue):
* Source/WebCore/css/ComputedStyleExtractor.cpp:
(WebCore::valueForPosition):
(WebCore::isAuto):
(WebCore::isNormal):
(WebCore::valueForPathOperation):
* Source/WebCore/platform/LengthFunctions.cpp:
(WebCore::pointForLengthPoint):
(WebCore::floatPointForLengthPoint):
* Source/WebCore/platform/LengthPoint.cpp:
(WebCore::operator<<):
* Source/WebCore/platform/LengthPoint.h:
(WebCore::LengthPoint::LengthPoint):
(WebCore::LengthPoint::isZero const):
(WebCore::blend):
(WebCore::LengthPoint::setX): Deleted.
(WebCore::LengthPoint::x const): Deleted.
(WebCore::LengthPoint::setY): Deleted.
(WebCore::LengthPoint::y const): Deleted.
* Source/WebCore/platform/animation/AcceleratedEffectValues.cpp:
(WebCore::nonCalculatedLengthPoint):
(WebCore::AcceleratedEffectValues::AcceleratedEffectValues):
* Source/WebCore/rendering/MotionPath.cpp:
(WebCore::MotionPath::motionPathDataForRenderer):
(WebCore::MotionPath::applyMotionPathTransform):
* Source/WebCore/rendering/RenderReplaced.cpp:
(WebCore::RenderReplaced::paint):
* Source/WebCore/rendering/style/ShadowData.cpp:
(WebCore::ShadowData::ShadowData):
* Source/WebCore/rendering/style/ShadowData.h:
(WebCore::ShadowData::ShadowData):
(WebCore::ShadowData::x const):
(WebCore::ShadowData::y const):
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

Canonical link: https://commits.webkit.org/284009@main
